x/stakeibc/keeper/msg_server_redeem_stake.go

```diff
@@ -84,20 +80,36 @@ func (k msgServer) RedeemStake(goCtx context.Context, msg *types.MsgRedeemStake)
 	if balance.Amount.LT(msg.Amount) {
 		return nil, errorsmod.Wrapf(sdkerrors.ErrInvalidCoins, "balance is lower than redemption amount. redemption amount: %v, balance %v: ", msg.Amount, balance.Amount)
 	}
-	// UNBONDING RECORD KEEPING
-	userRedemptionRecord := recordstypes.UserRedemptionRecord{
-		Id:          redemptionId,
-		Sender:      senderAddr,
-		Receiver:    msg.Receiver,
-		Amount:      nativeAmount,
-		Denom:       hostZone.HostDenom,
-		HostZoneId:  hostZone.ChainId,
-		EpochNumber: epochTracker.EpochNumber,
-		// claimIsPending represents whether a redemption is currently being claimed,
-		// contingent on the host zone unbonding having status CLAIMABLE
-		ClaimIsPending: false,
+
+	// ----------------- UNBONDING RECORD KEEPING -----------------
+	// Fetch the record
+	senderAddr := sender.String()
+	redemptionId := recordstypes.UserRedemptionRecordKeyFormatter(hostZone.ChainId, epochTracker.EpochNumber, senderAddr)
+	userRedemptionRecord, userHasRedeemedThisEpoch := k.RecordsKeeper.GetUserRedemptionRecord(ctx, redemptionId)
+	if userHasRedeemedThisEpoch {
+		k.Logger(ctx).Info(fmt.Sprintf("UserRedemptionRecord found for %s", redemptionId))
+		// Add the unbonded amount to the UserRedemptionRecord
+		// The record is set below
+		userRedemptionRecord.Amount.Add(nativeAmount)
```

Comment:
need to set the amount on the record here
```suggestion
		userRedemptionRecord.Amount = userRedemptionRecord.Amount.Add(nativeAmount)
```

Reply:
Good catch!